site/public/js/details.js

```diff
@@ -110,22 +110,126 @@ function collapseAllSections() {
+    }
+
+    // filter student who withdrawned from this course
+    filterWithdrawnCheckbox.on('change', () => {
+        if (Cookies.get('filter_student') === 'false') {
+            $('[data-student="electronic-grade-withdrawn"]').hide();
+            Cookies.set('filter_student', true);
+        }
+        else {
+            $('[data-student="electronic-grade-withdrawn"]').show();
+            Cookies.set('filter_student', false);
+        }
+    });
+}
+
 function inquiry_update() {
-    const button = document.getElementById('inquiryButton');
+    const check_inquiry = $('[data-testid="toggle-grade-inquiry"]');
     const status = Cookies.get('inquiry_status');
-
     if (status === 'on') {
         $('.grade-button').each(function () {
             if (typeof $(this).attr('data-grade-inquiry') === 'undefined') {
-                $(this).closest('.grade-table').hide(); // hide gradeable items without active inquiries
+                $(this).closest('.grade-table').hide();
+                check_inquiry.prop('checked', true);
             }
         });
-        button.textContent = 'Grade Inquiry Only: On';
+    }
+    check_inquiry.on('change', () => {
+        if (status === 'on') {
+            Cookies.set('inquiry_status', 'off');
+        }
+        else {
+            Cookies.set('inquiry_status', 'on');
+        }
+        location.reload();
+    });
+}
+
+function switch_view() {
+    // true means view all, false means view assigned section
+    const view_status = $('[data-testid="view-sections"]');
+    if (Cookies.get('view') === undefined) {
+        Cookies.set('view', 'assigned', { path: '/' });
+        localStorage.setItem('general-setting-navigate-assigned-students-only', 'true');
+        location.reload();
+    }
+    else if (Cookies.get('view') === 'all') {
+        view_status.prop('checked', false);
     }
     else {
-        $('.grade-button').each(function () {
-            $(this).closest('.grade-table').show(); // show all gradeable items
-        });
-        button.textContent = 'Grade Inquiry Only: Off';
+        view_status.prop('checked', true);
+    }
+    view_status.on('change', () => {
+        if (Cookies.get('view') === 'all') {
+            Cookies.set('view', 'assigned', { path: '/' });
+            localStorage.setItem('general-setting-navigate-assigned-students-only', 'true');
+        }
+        else {
+            Cookies.set('view', 'all', { path: '/' });
+            localStorage.setItem('general-setting-navigate-assigned-students-only', 'false');
+        }
+        location.reload();
+    });
+}
+
+function change_anon() {
+    const gradeable_id = document.getElementById('toggle-anon-button').getAttribute('data-gradeable-id');
+    const anon_status = $('[data-testid="toggle-anon-button"]');
+    Cookies.set(`default_anon_mode_${gradeable_id}_override`, 'on');
+    if (Cookies.get(`anon_mode_${gradeable_id}`) === undefined) {
+        Cookies.set(`anon_mode_${gradeable_id}`, 'off');
```

Comment:
Do you need to set the status of `anon_status` here?  I guess if it defaults to unchecked that's good enough?